hedera-node/src/main/java/com/hedera/services/ledger/TransferLogic.java

```diff
@@ -137,23 +153,35 @@ public void doZeroSum(final List<BalanceChange> changes) {
         if (validity == OK) {
             adjustBalancesAndAllowances(changes);
             if (autoCreationFee > 0) {
-                payFunding(autoCreationFee);
+                payAutoCreationFee(autoCreationFee);
                 autoCreationLogic.submitRecordsTo(recordsHistorian);
+                autoCreationLogic.clearTokenAliasMap();
             }
         } else {
             dropTokenChanges(sideEffectsTracker, nftsLedger, accountsLedger, tokenRelsLedger);
             if (autoCreationLogic != null && autoCreationLogic.reclaimPendingAliases()) {
                 accountsLedger.undoCreations();
+                autoCreationLogic.clearTokenAliasMap();
             }
             throw new InvalidTransactionException(validity);
         }
     }
 
-    private void payFunding(final long autoCreationFee) {
+    private boolean shouldAutoCreate(final BalanceChange change) {
+        return change.hasNonEmptyAlias()
+                || (change.isForNft() && change.hasNonEmptyCounterPartyAlias());
+    }
+
+    private void payAutoCreationFee(final long autoCreationFee) {
         final var funding = dynamicProperties.fundingAccount();
         final var fundingBalance = (long) accountsLedger.get(funding, BALANCE);
         final var newFundingBalance = fundingBalance + autoCreationFee;
         accountsLedger.set(funding, BALANCE, newFundingBalance);
+
+        // deduct the auto creation fee from payer of the transaction
```

Comment:
We should actually use `FeeDistribution` here, right?

Reply:
Yes that make sense. I think its missed during staking changes